pkg/syncer/metadatasyncer.go

```diff
@@ -182,17 +183,13 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
 				return err
 			}
 			if len(clusterComputeResourceMoIds) > 0 {
```

Comment:
Do we still need this check here? If it is workload flavor and TKGs-HA is enabled, we can assign `supervisor-id` to  `clusterIDforVolumeMetadata`

Reply:
moved assignment for `clusterIDforVolumeMetadata` out

```
			if len(clusterComputeResourceMoIds) > 0 {
				if configInfo.Cfg.Global.SupervisorID == "" {
					return logger.LogNewError(log, "supervisor-id is not set in the vsphere-config-secret")
				}
			}
			clusterIDforVolumeMetadata = configInfo.Cfg.Global.SupervisorID
```